pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -0,0 +1,255 @@
+package guard
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/informers"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	policyclientv1 "k8s.io/client-go/kubernetes/typed/policy/v1"
+	corelisterv1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/guard/bindata"
+)
+
+// GuardController is a controller that watches amount of static pods on master nodes and
+// renders guard pods with a pdb to keep maxUnavailable to be at most 1
+type GuardController struct {
+	targetNamespace, podResourcePrefix string
+	operatorName                       string
+	readyzPort                         string
+
+	nodeLister corelisterv1.NodeLister
+	podGetter  corev1client.PodsGetter
+	pdbGetter  policyclientv1.PodDisruptionBudgetsGetter
+
+	// installerPodImageFn returns the image name for the installer pod
+	installerPodImageFn   func() string
+	createConditionalFunc resourceapply.ConditionalFunction
+	deleteConditionalFunc resourceapply.ConditionalFunction
+}
+
+func NewGuardController(
+	targetNamespace, podResourcePrefix string,
+	operatorName string,
+	readyzPort string,
+	kubeInformersForTargetNamespace informers.SharedInformerFactory,
+	kubeInformersClusterScoped informers.SharedInformerFactory,
+	podGetter corev1client.PodsGetter,
+	pdbGetter policyclientv1.PodDisruptionBudgetsGetter,
+	eventRecorder events.Recorder,
+	createConditionalFunc resourceapply.ConditionalFunction,
+	deleteConditionalFunc resourceapply.ConditionalFunction,
+) factory.Controller {
+	c := &GuardController{
+		targetNamespace:       targetNamespace,
+		podResourcePrefix:     podResourcePrefix,
+		operatorName:          operatorName,
+		readyzPort:            readyzPort,
+		nodeLister:            kubeInformersClusterScoped.Core().V1().Nodes().Lister(),
+		podGetter:             podGetter,
+		pdbGetter:             pdbGetter,
+		installerPodImageFn:   getInstallerPodImageFromEnv,
+		createConditionalFunc: createConditionalFunc,
+		deleteConditionalFunc: deleteConditionalFunc,
+	}
+
+	return factory.New().WithInformers(
+		kubeInformersForTargetNamespace.Core().V1().Pods().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().Nodes().Informer(),
```

Comment:
one of these is wrong.  nodes are clusterscoped, so namespace scoping doesn't work.  and you want to be sure you only watch pods in your namespace.

Reply:
`kubeInformersForTargetNamespace.Core().V1().Nodes().Informer()` replaced with `kubeInformersClusterScoped.Core().V1().Nodes().Informer()`